WooCommerce/Classes/ViewRelated/Top Banner/TopBannerView.swift

```diff
@@ -122,19 +111,63 @@ private extension TopBannerView {
         actionButton.setTitle(viewModel.actionButtonTitle, for: .normal)
     }
 
+    func configureTopButton(viewModel: TopBannerViewModel, onContentView contentView: UIView) {
+        switch viewModel.topButton {
+        case .chevron:
+            updateExpandCollapseState(isExpanded: isExpanded)
+            expandCollapseButton.tintColor = .textSubtle
+
+            let tapGesture = UITapGestureRecognizer(target: self, action: #selector(onExpandCollapseButtonTapped))
+            tapGesture.cancelsTouchesInView = false
+            contentView.addGestureRecognizer(tapGesture)
+
+        case .dismiss:
+            dismissButton.setImage(UIImage.gridicon(.cross, size: CGSize(width: 24, height: 24)), for: .normal)
+            dismissButton.tintColor = .textSubtle
+            dismissButton.addTarget(self, action: #selector(onDismissButtonTapped), for: .touchUpInside)
+        }
+    }
+
     func configureBackground() {
         backgroundColor = .systemColor(.secondarySystemGroupedBackground)
     }
 
-    func createContentView() -> UIView {
-        let actionButon = isActionEnabled ? dismissButton : expandCollapseButton
-        let titleStackView = UIStackView(arrangedSubviews: [titleLabel, actionButon])
+    func createMainStackView(with viewModel: TopBannerViewModel) -> UIStackView {
+        let iconInformationStackView = createIconInformationStackView(with: viewModel)
+        let mainStackView = UIStackView(arrangedSubviews: [iconInformationStackView, createBorderView()])
+        if isActionEnabled {
+            mainStackView.addArrangedSubview(actionStackView)
+        }
+
+        mainStackView.translatesAutoresizingMaskIntoConstraints = false
+        mainStackView.axis = .vertical
+        return mainStackView
+    }
+
+    func createIconInformationStackView(with viewModel: TopBannerViewModel) -> UIStackView {
+        let informationStackView = createInformationStackView(with: viewModel)
+        let iconInformationStackView = UIStackView(arrangedSubviews: [iconImageView, informationStackView])
+
+        iconInformationStackView.translatesAutoresizingMaskIntoConstraints = false
+        iconInformationStackView.axis = .horizontal
+        iconInformationStackView.spacing = 16
+        iconInformationStackView.alignment = .leading
+        iconInformationStackView.layoutMargins = .init(top: 16, left: 16, bottom: 16, right: 16)
+        iconInformationStackView.isLayoutMarginsRelativeArrangement = true
```

Comment:
nit: just curious, was `isLayoutMarginsRelativeArrangement` set to fix some UI issue?

Reply:
This is to make the content laid-out inside `iconInformationStackView` to respect the provided layout margins.